### PR TITLE
Fix compute instance iam asset name

### PR DESCRIPTION
**Pull request by iyabchen**
Fix compute instance iam asset name

Asset name was expecting `instance`, but compute instance iam uses instance_name as attribute name. 

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by melinath — COMMENTED** [review]
This case should be handled by the resource_converter_iam template; ideally we'd be able to fix the bug instead of making this handwritten. The resource converter template has access to `instance_name` being the parent_resource_attribute - this seems like maybe a bug in how the `asset_name_template` is built when there isn't an explicit import_format. https://github.com/GoogleCloudPlatform/magic-modules/blob/0f8b95f486cf0373e635b73d57cbf63db6337b8e/mmv1/templates/validator/resource_converter_iam.go.erb#L28

**Comment by iyabchen** [comment]
> This case should be handled by the resource_converter_iam template; ideally we'd be able to fix the bug instead of making this handwritten. The resource converter template has access to `instance_name` being the parent_resource_attribute - this seems like maybe a bug in how the `asset_name_template` is built when there isn't an explicit import_format.
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/0f8b95f486cf0373e635b73d57cbf63db6337b8e/mmv1/templates/validator/resource_converter_iam.go.erb#L28

Should we add an import_format for that resource, or add a logic like 
```
  import_format = object.iam_policy.import_format || object.import_format
  if import_format.nil? || import_format.empty?
    ...
  else
    asset_name_template = '//' + product_backend_name.downcase + '.googleapis.com/' + import_format[0].gsub('{{name}}', "{{#{parent_resource_name}}}").gsub('{{instance}}', "{{#{parent_resource_name}}}").gsub('%', '')
  end
```

**Comment by melinath** [comment]
@iyabchen we shouldn't need to add an import format and we shouldn't need to make a special case - we should just need to change the first case (import format being nil) to correctly reference `parent_resource_name` when building the asset name template, which will probably look similar to what currently happens in the second case.

File: mmv1/templates/validator/resource_converter_iam.go.erb [hunk 1]
```diff
@@ -21,12 +21,13 @@
...
   resource_uri = object.iam_policy.base_url || object.self_link_uri
   parent_resource_name = object.iam_policy.parent_resource_attribute || object.name.underscore
+  asset_resource_name = parent_resource_name.downcase || object.name.downcase
   resource_params = extract_identifiers(resource_uri.gsub('{{name}}', "{{#{parent_resource_name}}}"))
   resource_params.delete('project')
 
   import_format = object.iam_policy.import_format || object.import_format
   if import_format.nil? || import_format.empty?
-    asset_name_template = '//' + product_backend_name.downcase + '.googleapis.com/' + (object.cai_base_url || object.self_link || object.base_url) + '/{{' + object.name.downcase + '}}'
+    asset_name_template = '//' + product_backend_name.downcase + '.googleapis.com/' + (object.cai_base_url || object.self_link || object.base_url) + '/{{' + asset_resource_name + '}}'
```
Comment (melinath) [inline mmv1/templates/validator/resource_converter_iam.go.erb:30]:
```suggestion
    asset_name_template = '//' + product_backend_name.downcase + '.googleapis.com/' + (object.cai_base_url || object.self_link || object.base_url) + '/{{' + parent_resource_name + '}}'
```

File: mmv1/templates/validator/resource_converter_iam.go.erb [hunk 2]
```diff
@@ -21,12 +21,13 @@
 
   resource_uri = object.iam_policy.base_url || object.self_link_uri
   parent_resource_name = object.iam_policy.parent_resource_attribute || object.name.underscore
```
Comment (melinath) [inline mmv1/templates/validator/resource_converter_iam.go.erb:23]:
```suggestion
  parent_resource_name = object.iam_policy.parent_resource_attribute.downcase || object.name.underscore
```

It looks like `underscore` includes a call to `.downcase` so I think setting `asset_resource name` may be duplicative (except for downcasing parent_resource_name in the case where it's not pulled from object.name - I'm not sure whether we need to downcase that? But it probably doesn't hurt either way.)

**Review by melinath — CHANGES_REQUESTED** [review]
💯 this is looking good - it fixes a few other iam resources as well! I think it might be possible to simplify the logic a bit? LGTM otherwise.

File: mmv1/templates/validator/resource_converter_iam.go.erb (hunk 1 above)
Comment (iyabchen) [reply mmv1/templates/validator/resource_converter_iam.go.erb]: Done. parent_resource_name alone works, so no need to `||  object.name`.

**Review by melinath — APPROVED** [review]

**Pull request merged**